### PR TITLE
Base Button

### PR DESCRIPTION
**Pull request by tenadopt**
Add Base Button

### CODE REVIEW

File: dist/components/.DS_Store
Comment (shev7) [inline dist/components/.DS_Store:1]: add to ```.gitignore```

File: src/components/BaseButton/BaseButton.module.scss [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    &.text {
+        border: 0;
+        &:active {
+            @include active-state;
+        }
+        &.disabled {
+            @include disabled-state;
+            background: none; // Specific to text variant
```
Comment (shev7) [inline src/components/BaseButton/BaseButton.module.scss]: you should use ```:disabled ``` pseudoclass to style button  or any other element in disabled state

File: src/components/BaseButton/BaseButton.tsx [hunk 2]
```diff
@@ -1,20 +1,36 @@
...
-                                                 ...props
-                                             }) => {
+const cx = cnBind.bind(styles);
+
+const BaseButton: FC<BaseButtonProps> = ({ variant, size, ...props }) => {
+    const btn = cx('button', props.className, {
+        disabled: props.disabled,
+        small: size === 'small',
+        medium: size === 'medium',
+        large: size === 'large',
```
Comment (shev7) [inline src/components/BaseButton/BaseButton.tsx]: you could pass size as a classname to button, in this case you won't need to specify condition for each size separately

File: src/components/BaseButton/BaseButton.tsx [hunk 3]
```diff
@@ -1,20 +1,36 @@
...
+        text: variant === 'text',
+        outlined: variant === 'outlined',
+        contained: variant === 'contained',
```
Comment (shev7) [inline src/components/BaseButton/BaseButton.tsx]: the same with variant

File: src/components/BaseButton/BaseButton.tsx [hunk 4]
```diff
@@ -1,20 +1,36 @@
...
 
-const BaseButton: FC<BaseButtonProps> = ({
-                                                 children,
-                                                 color,
-                                                 ...props
-                                             }) => {
+const cx = cnBind.bind(styles);
+
+const BaseButton: FC<BaseButtonProps> = ({ variant, size, ...props }) => {
+    const btn = cx('button', props.className, {
```
Comment (shev7) [inline src/components/BaseButton/BaseButton.tsx]:
```suggestion
    const btn = cx('button', props.className, size, variant, {
```
if you pass size and variant like this you want need to specify condition for each size and for each variant
imagine if you will have more 3 variants

File: src/components/BaseButton/BaseButton.tsx [hunk 5]
```diff
@@ -1,20 +1,36 @@
...
+        disabled: props.disabled,
```
Comment (shev7) [inline src/components/BaseButton/BaseButton.tsx]: use pseudo class instead

**Review by shev7 — CHANGES_REQUESTED** [review]

File: src/components/BaseButton/BaseButton.tsx (hunk 2 above)
Comment (tenadopt) [reply src/components/BaseButton/BaseButton.tsx]: done

File: src/components/BaseButton/BaseButton.tsx (hunk 3 above)
Comment (tenadopt) [reply src/components/BaseButton/BaseButton.tsx]: done

File: src/components/BaseButton/BaseButton.tsx (hunk 4 above)
Comment (tenadopt) [reply src/components/BaseButton/BaseButton.tsx]: done

File: src/components/BaseButton/BaseButton.tsx (hunk 5 above)
Comment (tenadopt) [reply src/components/BaseButton/BaseButton.tsx]: done

File: src/components/BaseButton/BaseButton.tsx [hunk 6]
```diff
@@ -1,20 +1,31 @@
...
 
-const BaseButton: FC<BaseButtonProps> = ({
-                                                 children,
-                                                 color,
-                                                 ...props
-                                             }) => {
+const cx = cnBind.bind(styles);
+
+/** Custom Button Component */
+const BaseButton = ({ variant, size, ...props }): BaseButtonProps => {
```
Comment (shev7) [inline src/components/BaseButton/BaseButton.tsx:15]:
```suggestion
const BaseButton = ({ variant, size, ...props }: BaseButtonProps) => {
```

**Review by shev7 — APPROVED** [review]

**Pull request merged**